### PR TITLE
feat: add prompts component

### PR DESCRIPTION
**Pull request by lyjeileen**
## Changes
- Add prompts component (styles are customizable.)
- Update messageSpace to display prompts
- Update theme for active state

## Screenshot/Video
Screenshot 2024-12-06 at 4 52 11 PM

Dec-06-2024 13-58-42

Screenshot 2024-12-07 at 3 26 04 PM

Screenshot 2024-12-06 at 4 52 29 PM

### CODE REVIEW

File: src/components/types.ts [hunk 1]
```diff
@@ -324,6 +324,17 @@ export interface QuestionFormat extends DataFormat {
...
+  /** A list of prompt strings for users to select from. These prompts will be displayed as interactive elements in the UI. */
+  prompts: string[]
+  /** An optional className to apply to the prompts container. */
+  className?: string
+}
+
+export interface PromptsProps
+  extends PromptsFormat,
+    Omit<ConversationProps, 'messageId'> {}
+
```
Comment (Shiti) [inline src/components/types.ts:337]: Display related properties like `className` etc should not be part of the `format` but component `props`
Also, I think its ok to include `messageId`. The component may not use it right now.

```suggestion
export interface PromptsFormat {
  /** A list of prompt strings for users to select from. These prompts will be displayed as interactive elements in the UI. */
  prompts: string[]
}

export interface PromptsProps
  extends PromptsFormat, ConversationProps {
      /** An optional className to apply to the prompts container. */
  className?: string
    }
    

```

Comment (Shiti) [reply src/components/types.ts:337]: I noticed the format is also missing the position but this is referred to in messageSpace

**Review by Shiti — CHANGES_REQUESTED** [review]

**Comment by Shiti** [comment]
@lyjeileen could you add a separate test for messageSpace and prompts display ? Maybe also test that only the latest prompts are shown in the message space

File: src/components/types.ts (hunk 1 above)
Comment (lyjeileen) [reply src/components/types.ts:337]: We are not passing position to the prompts component, so I only mentioned `position` in the storybook. What do you think?

Comment (Shiti) [reply src/components/types.ts:337]: The format dictates what fields are expected from a message of that format on the websocket. Even if the field is not used within the component, it is being used by messageSpace. Adding it here and specifying the expected values would be cleaner when integrating with the library.

**Review by Shiti — APPROVED** [review]

**Pull request merged**